strings/manacher_algorithm.cpp

```diff
@@ -0,0 +1,171 @@
+/**
+ * @file
+ * @brief Implementation of [Manacher's
+ * Algorithm](https://en.wikipedia.org/wiki/Longest_palindromic_substring)
+ * @details
+ * Manacher's Algorithm is used to find longest palindromic substring within a
+ * string in O(n) time. It exploits the property of a palindrome that its first
+ * half is symmetric to the last half, and thus if the first half is a
+ * palindrome, then last half is also a palindrome.
+ * @author [Riti Kumari](https://github.com/riti2409)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <vector>    /// for std::vector STL
+#ifdef _MSC_VER
+#include <string>  /// for string (required for MS Visual C++)
+#else
+#include <cstring>  /// for string
+#endif
+
+/**
+ * @namespace strings
+ * @brief Algorithms with strings
+ */
+namespace strings {
+/**
+ * @namespace manacher
+ * @brief Functions for [Manacher's
+ * Algorithm](https://en.wikipedia.org/wiki/Longest_palindromic_substring)
+ * implementation
+ */
+namespace manacher {
+/**
+ * @brief A function that implements Manacher's algorithm
+ * @param prototype is the string where algorithm finds a palindromic substring.
+ * This string can contain any character except `@` `#` `&`
+ * @returns the largest palindromic substring
+ */
+std::string manacher(const std::string &prototype) {
+    if (prototype.size() > 0) {
+        // stuffing characters between the input string to handle cases with
+        // even length palindrome
+        std::string stuffed_string = "";
+        for (auto str : prototype) {
+            stuffed_string += str;
+            stuffed_string += "#";
+        }
+        stuffed_string = "@#" + stuffed_string + "&";
+
+        std::vector<int> palindrome_max_half_length(
+            stuffed_string.size(),
+            0);  // int palindrome_max_half_length[stuffed_string.size()] = {0};
+                 // this array will consist of largest possible half length of
+                 // palindrome centered at index (say i with respect to the
+                 // stuffed string). This value will be lower bound of half
+                 // length since single character is a palindrome in itself.
+
+        int bigger_center =
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). 🙂

Reply:
@Panquesito7 Sure, Updated